onnxscript/rewriter/broadcast_to_matmul.py

```diff
@@ -108,9 +108,11 @@ def check_if_need_reshape(input_a, input_b, shape_c, **_) -> bool:
     broadcast_matmul_output_shape = (
         longer_shape[: -len(shorter_shape)] + broadcast_matmul_output_shape
     )
-    if mimic_matmul_broadcast_behavior and dim_b == 2:
+    if mimic_matmul_broadcast_behavior and dim_b == 2 and input_b_shape[-1] == 1:
+        # If input_b is expanded to 2-D, then we need to remove the last dimension
         broadcast_matmul_output_shape = broadcast_matmul_output_shape[:-1]
-    if mimic_matmul_broadcast_behavior and dim_a == 2:
+    if mimic_matmul_broadcast_behavior and dim_a == 2 and input_a_shape[0] == 1:
+        # If input_a is expanded to 2-D, then we need to remove the duplicated dimension
```

Comment:
What does "duplicated dimension" mean? Was it a sized 1 dim that was broadcasted?

Reply:
Changed it to the first dim of input_a, which is -2nd dim of the mimic output shape.